catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/IngestCommand.java

```diff
@@ -201,128 +218,72 @@ protected Object executeWithSubject() throws Exception {
                 blockingQueue,
                 rejectedExecutionHandler);
 
-        if (inputFile.isDirectory()) {
-            final long start = System.currentTimeMillis();
-
-            List<File> files = getFiles(inputFile);
-
-            fileCount.getAndSet(files.size());
+        final long start = System.currentTimeMillis();
 
-            printProgressAndFlush(start, fileCount.get(), ingestCount.get() + ignoreCount.get());
+        printProgressAndFlush(start, fileCount.get(), 0);
 
-            List<List<File>> partition = Lists.partition(files, batchSize);
-
-            try {
-                for (List<File> batch : partition) {
-                    IngestRunnable ingestRunnable = new IngestRunnable(start,
-                            catalog,
-                            ignoreList,
-                            batch);
-                    executorService.submit(ingestRunnable);
-                }
-            } finally {
-                executorService.shutdown();
-            }
-
-            while (!executorService.isTerminated()) {
-                try {
-                    TimeUnit.MILLISECONDS.sleep(100);
-                } catch (InterruptedException e) {
-                    // ignore
-                }
+        executorService.submit(new Runnable() {
```

Comment:
The executor that you're using at this point seems like it should be a single threaded executor and rather than creating a single threaded executor down below, you want that one to actually be the multi-threaded executor that this chunk is using. As it is, you're running 1 thread to build a queue and 1 thread to process batches. The CatalogFramework could handle multiple batches at a time, however, so flipping these executors would likely speed things up quite a bit.